src/api/root/handler.js

```diff
@@ -0,0 +1,184 @@
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * Lewis Daly <lewis@vesselstech.com>
+ --------------
+ ******/
+
+const P = require('bluebird')
+
+const packageJson = require('../../../package.json')
+const Util = require('../../handlers/lib/utility')
+const Kafka = require('../../handlers/lib/kafka/index')
+const Enum = require('../../lib/enum')
+const MigrationLockModel = require('../../models/misc/migrationLock')
+
+const TransferEventType = Enum.transferEventType
+const TransferEventAction = Enum.transferEventAction
+
+const statusEnum = {
+  OK: 'OK',
+  DOWN: 'DOWN'
+}
+
+const serviceName = {
+  datastore: 'datastore',
+  broker: 'broker',
+  sidecar: 'sidecar',
+  cache: 'cache'
+}
+
+/**
+ * getHealth
+ *
+ * Gets the health status of the service.
+ *
+ *
+ * @example:
+ * returns the health of related sub-services:
+ * {
+ *   "status": "OK",
+ *   "uptime": 0,
+ *   "started": "2019-05-31T05:09:25.409Z",
+ *   "versionNumber": "5.2.3",
+ *   "services": [
+ *     {
+ *       "name": "datastore",
+ *       "status": "OK"
+ *     }
+ *   ]
+ * }
+ *
+ *
+ * @param {*} request - the request object
+ * @param {*} h       - the handler object
+ */
+const getHealth = async function (request, h) {
+  // Default values
+  let statusCode = 200
+  let status = statusEnum.OK
+  let isHealthy = true
+  let services = []
+  let subServices = {}
+
+  const uptime = process.uptime() // in seconds by default
+  const startTimeDate = new Date(Date.now() - uptime)
+  const startTime = startTimeDate.toISOString()
+  const versionNumber = packageJson.version
+
+  try {
+    services = [
+      await getSubServiceHealth(serviceName.datastore),
+      await getSubServiceHealth(serviceName.broker)
+    ]
+    isHealthy = evaluateServiceHealth(services)
+    subServices = {
+      services
+    }
+  } catch (err) {
+    isHealthy = false
+  }
+
+  if (!isHealthy) {
+    statusCode = 502
+    status = statusEnum.DOWN
+  }
+
+  const response = {
+    status,
+    uptime,
+    startTime,
+    versionNumber,
+    ...subServices
+  }
+  return h.response(response).code(statusCode)
+}
+
+/**
+ * getSubServiceHealth
+ *
+ * Gets the health status of a sub-service. Takes advantage of existing
+ * connections to the service (doesn't implement a ping or )
+ */
+const getSubServiceHealth = async function (serviceName) {
+  switch (serviceName) {
+    case 'datastore': {
+      const isLocked = await MigrationLockModel.getIsMigrationLocked()
+
+      return {
+        name: serviceName,
+        status: isLocked ? statusEnum.DOWN : statusEnum.OK
+      }
+    }
+    case 'broker': {
+      /*
+        Ensure that all topics for consumers and producers are available
+        Otherwise consider Kafka DOWN
+      */
+      const consumerTopics = [
+        Util.transformGeneralTopicName(TransferEventType.TRANSFER, TransferEventAction.PREPARE),
```

Comment:
@mdebarros @lewisdaly I believe this will fail when the API is run standalone. This needs to be taken into consideration for the API as well as the API for the Kafka handlers. This will in turn effect the getSubServiceHealth and the evaluateServiceHealth functions. I believe we need to revisit the way this check is done to something more generic

Reply:
Sure. I think I need to get a better understanding of running the api in standalone first.

Do we have any tests (integration or otherwise) that test the endpoints in standalone?

Reply:
Yes @lewisdaly we do have integration tests in test/integration/domain/participant/index.test.js
But i would also look where the initialise function in src/shared/setup.js is called from and the different ways that it is started up. If you need any assistance at any point please feel free to message me or @mdebarros on slack.